scripts/build_failure_data_collection.py

```diff
@@ -0,0 +1,294 @@
+    backed_out_revisions = []
+
+    for bug_id in bug_ids:
+        bug_id_commits = bug_commits.get(bug_id, None)
+        backing_out_commit = find_backing_out_commit(bug_id_commits, hg_client)
+
+        if not backing_out_commit:
+            continue
+
+        logger.info("Backing out commit found!")
+
+        # commit = {}
+
+        # commit["desc"] = next(
+        #     (
+        #         c["desc"]
+        #         for c in bug_id_commits
+        #         if any(
+        #             c["node"].startswith(node)
+        #             for node in backing_out_commit["backsout"]
+        #         )
+        #     ),
+        #     None,
+        # )
+
+        # if commit["desc"] is None:
+        #     continue
+
+        commits = [
+            {
+                "desc": c["desc"],
+            }
+            for c in bug_id_commits
+            if any(
+                c["node"].startswith(node) for node in backing_out_commit["backsout"]
+            )
+        ]
+
+        if commits is None:
+            continue
+
+        for commit in commits:
+            revision_id = repository.get_revision_id(commit)
+            backed_out_revisions.append(revision_id)
+
+    return backed_out_revisions
+
+
+def find_backing_out_commit(commits, hg_client):
+    logger.info("Finding backing out commit...")
+    if not commits:
+        return None
+
+    for commit in commits:
+        if not commit["backsout"]:
+            continue
+
+        desc = commit["desc"]
+        if (
+            "backed out" in desc.lower()
+            and "for causing" in desc.lower()
+            and "build" in desc.lower()
+        ):
+            return commit
+    return None
+
+
+def find_error_lines(index_client, queue_client, commit_node):
+    # FINAL STEPS
+    # 1. list the tasks
+    tasks = index_client.listTasks(f"gecko.v2.autoland.revision.{commit_node}.firefox")
+
+    if not tasks["tasks"]:
+        return []
+
+    # 2. get the task ID from one of the tasks (I think any is fine)
+    first_task_id = tasks["tasks"][0]["taskId"]
+
+    # 3. get the task group ID from the task ID
+    first_task = queue_client.task(first_task_id)
+    task_group_id = first_task["taskGroupId"]
+
+    # 4. extract the build task IDs from the task group ID
+    url = f"https://firefoxci.taskcluster-artifacts.net/{task_group_id}/0/public/label-to-taskid.json"
+    response = requests.get(url)
+    response.raise_for_status()
+    data = response.json()
+
+    build_tasks = set()
+
+    for label, taskId in data.items():
+        if label[:5] == "build":
+            build_tasks.add(taskId)
+
+    # 5. get failed tasks
+    failed_tasks = set()
+
+    for task in queue_client.listTaskGroup(task_group_id)["tasks"]:
+        if task["status"]["state"] == "failed":
+            failed_tasks.add(task["status"]["taskId"])
+
+    # 6. find intersection between build tasks and failed tasks
+    failed_build_tasks = list(build_tasks & failed_tasks)
```

Comment:
if you use treeherder db (search for fixed_by_commit) instead of bugzilla to find backouts you get most of this stuff for free.